File: mybatis-plus-join-test/src/main/java/com/github/yulichang/test/enums/Sex.java
package com.github.yulichang.test.enums;

import com.baomidou.mybatisplus.annotation.EnumValue;
import lombok.Getter;
import lombok.ToString;

@Getter
@ToString
public enum Sex {

    MAN(0, "男"),

    WOMAN(1, "女");

    @EnumValue
    private final int code;
    private final String des;

    Sex(int code, String des) {
        this.code = code;
        this.des = des;
    }

}

File: mybatis-plus-join-test/test-kotlin/src/main/kotlin/com/github/yulichang/test/kt/enums/Sex.kt
package com.github.yulichang.test.kt.enums

import com.baomidou.mybatisplus.annotation.EnumValue

@Suppress("unused")
enum class Sex(@field:EnumValue private val code: Int, private val des: String) {
    MAN(0, "男"),
    WOMAN(1, "女")
}
